packages/data-context/src/sources/CloudDataSource.ts

```diff
@@ -174,6 +153,22 @@ export class CloudDataSource {
     return `${print(config.document)}-${stringifyVariables(config.variables)}`
   }
 
+  #formatWithErrors (data: OperationResult<any, any>) {
+    // If we receive a 401 from the dashboard, we need to logout the user
+    if (data.error?.response?.status === 401) {
+      this.params.logout()
+    }
+
+    if (data.error?.graphQLErrors[0]) {
+      this.params.onError(data.error?.graphQLErrors[0])
+    }
```

Comment:
I think we can get rid of this and the `onError` we pass to `CloudDataSource` entirely

Reply:
Updated in [cd05ee9](https://github.com/cypress-io/cypress/pull/21542/commits/cd05ee95576355255aa228cbac2f2ff5b58a434b)